UPGRADE.md

```diff
@@ -1413,6 +1413,49 @@ const PostShow = () => (
 };
 ```
 
+## No More Props Injection In `<AppBar>`
+
+When a React element was provided as the `userMenu` prop, the `<AppBar>` used to clone it and inject the `logout` prop (a React element). This is no longer the case and if you provided a custom user menu, you now have to include the logout yourself.
+
+```diff
+-import { UserMenu } from 'react-admin';
++import { Logout, UserMenu } from 'react-admin';
+
+-export const MyUserMenu = (props) => (
++export const MyUserMenu = () => (
+-    <UserMenu {...props}>
++    <UserMenu logout={<Logout />}>
```

Comment:
Isn't it better to invite users to add the logout element as a child of the UserMenu?

Reply:
It would probably be better. Requires a bit more refactoring to get rid of the logout prop alltogether